src/shared_state/conflicts.md

```diff
@@ -0,0 +1,145 @@
+It is ultimately up to an app's developers whether they want to support such use-cases.
+
+## Partial ordering
+
+In the example above, Alejo and Binita can not eliminate the possibility of messages being sent at the same time, and thus cannot guarantee that all messages will have a consistent ordering.
+Even so, it is possible for at least some of these messages to have a consistent ordering.
+When this is achieved, the collection of all their messages can be said to be [**partially ordered**](https://en.wikipedia.org/wiki/Partially_ordered_set).
+
+There are a number of techniques through which this can be accomplished, generally using either a [logical clock](https://en.wikipedia.org/wiki/Logical_clock) (commonly using [Lamport timestamps](https://en.wikipedia.org/wiki/Lamport_timestamp)) or by structuring messages in a [directed acyclic graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph) using references to the immediately preceding messages.
+
+Computers can have inaccurate physical clocks, so relying on them to order messages can be problematic, potentially resulting in messages that appear to come from the impossibly distant past or future.
+Instead, these more advanced techniques depend on nothing but the system itself to detect when two messages were sent and received in the correct sequence.
+
+Lamport timestamps, for example, set the counter of the current message to one more than the previously greatest timestamp.
+If Alice and Bob both tried to repeatedly set the description for a group, it might play out like so:
+
+1. Alejo sets the description using a message (A) with an id `0`
+2. Binita receives this message, but wants to correct a typo, so she sends an update (B) with id `1`. Alejo can determine that this was sent and received in the right order, so this should supercede his own description rather than be treated as a conflict
+3. They both decide to add some emojis to the description, and do so in parallel (messages C and D), but with different emojis. Each believes their message to correctly have the id `2`
+4. Before either of the previous messages have been delivered, Binita sends one more update (E) to add another emoji using id `3`
+5. Alejo and Binita receive each others' messages:
+    * because C and D both have id `2`, Binita can conclude that Alejo was not aware of her message when he authored his
+    * Alejo can conclude the same of Binita's D, but cannot determine whether she was aware of his C when she sent E. Because of this uncertainty, they should both treat these messages as having been authored concurrently, resulting in a conflict
+6. Alejo decides to resolve this conflict, authoring a new description which includes the corrected typo and all emojis added by either of them. When Binita receives it will have clearly been sent after both C and E
```

Comment:
`When Binita receives it will..` maybe change to `When Binita receives it, it will..`